### PR TITLE
Use the fetch stack for stylesheets.

### LINKED ISSUE #13462: Use the fetch stack for stylesheets
I'm looking into this.

### PR DESCRIPTION
**Pull request by Ms2ger**
Fixes #13462.

---

This change is [Reviewable](https://reviewable.io/reviews/servo/servo/13575)

### CODE REVIEW

File: components/script/dom/htmllinkelement.rs [hunk 1]
```diff
@@ -256,10 +255,22 @@ impl HTMLLinkElement {
...
+        let request = RequestInit {
+            url: url.clone(),
+            type_: RequestType::Script,
+            destination: Destination::Style,
+            credentials_mode: CredentialsMode::Include,
+            origin: document.url().clone(),
+            pipeline_id: Some(self.global().r().pipeline_id()),
+            referrer_url: Some(document.url().clone()),
+            referrer_policy: referrer_policy,
+            .. RequestInit::default()
```
Comment (jdm) [inline components/script/dom/htmllinkelement.rs]: This should set `use_url_credentials` to true per https://html.spec.whatwg.org/multipage/infrastructure.html#create-a-potential-cors-request , as well as set mode to Cors.

File: components/script/dom/htmllinkelement.rs [hunk 2]
```diff
@@ -256,10 +255,22 @@ impl HTMLLinkElement {
...
         let referrer_policy = match self.RelList().Contains("noreferrer".into()) {
             true => Some(ReferrerPolicy::NoReferrer),
-            false => None,
+            false => document.get_referrer_policy(),
+        };
+
+        let request = RequestInit {
+            url: url.clone(),
+            type_: RequestType::Script,
+            destination: Destination::Style,
```
Comment (jdm) [inline components/script/dom/htmllinkelement.rs]: The spec says destination should be the empty string, but the table at https://fetch.spec.whatwg.org/#concept-request-type disagrees. I've filed https://github.com/whatwg/html/issues/1856.

File: components/script/dom/htmllinkelement.rs [hunk 3]
```diff
@@ -256,10 +255,22 @@ impl HTMLLinkElement {
 
         let referrer_policy = match self.RelList().Contains("noreferrer".into()) {
             true => Some(ReferrerPolicy::NoReferrer),
-            false => None,
+            false => document.get_referrer_policy(),
+        };
+
+        let request = RequestInit {
+            url: url.clone(),
+            type_: RequestType::Script,
```
Comment (jdm) [inline components/script/dom/htmllinkelement.rs]: The spec doesn't specify a type; the table at https://fetch.spec.whatwg.org/#concept-request-type says it should be style.

**Review by jdm — CHANGES_REQUESTED** [review]

**Comment by Ms2ger** [comment]
Done

**Pull request merged**